core-web/libs/sdk/experiments/src/lib/parser/parser.ts

```diff
@@ -0,0 +1,65 @@
+import { AssignedExperiments, ExperimentParsed } from '../models';
+
+/**
+ * This arrow function parses a given set of assigned experiments for analytics.
+ *
+ * This process involves iterating over the experiments, which are currently in the "Running" state as received from the DotCMS endpoint,
+ * analyzing each experiment's relevant data such as running ID, variant name, and look back window value.
+ * It also performs regular expression verification for both 'isExperimentPage' and 'isTargetPage' against the current URL.
+ *
+ * The parsed data is useful for tracking and understanding the user's interaction with the experiment-targeted components during their visit.
+ *
+ * @param {AssignedExperiments} data - This parameter represents the assigned experiments data received from DotCMS endpoint.
+ * Contains an object with experiments information.
+ *
+ * @param {Location} location - This parameter is the object representing the current location (URL) of the user.
+ * Mostly employed for matching the regular expressions to detect whether the current page is an 'ExperimentPage' or a 'TargetPage'.
+ *
+ * @returns {ExperimentParsed} - The function returns an object with the original URL and an array of each experiment's comprehensive detail.
+ * The return object is suitable for further analytical operations. Each experiment's detail includes the experiment ID, running ID, variant name,
+ * look back window value, and booleans that represent whether current URL is 'isExperimentPage' or 'isTargetPage' for the respective experiment.
+ */
+export const parseDataForAnalytics = (
```

Comment:
Is this parser exposed in the library API? Meaning... can a developer import this into their projects?

Reply:
No. All related to the Experiment only will be used by the `DotExperiment` Class.